### PR TITLE
new resource: azurerm_api_management_api_tag #14453

### PR DESCRIPTION
**Pull request by meilz381**
Fix for [#14453 ](https://github.com/hashicorp/terraform-provider-azurerm/issues/14453)

### CODE REVIEW

File: website/docs/r/api_management_api_tag.html.markdown [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+}
+```
+
+## Arguments Reference
+
+The following arguments are supported:
+
+* `api_id` - (Required) The ID of the API Management API. Changing this forces a new API Management API Tag to be created.
+
+* `tag_name` - (Required) The name of the tag. It must be known in the API Management instance. Changing this forces a new API Management API Tag to be created.
```
Comment (katbyte) [inline website/docs/r/api_management_api_tag.html.markdown:50]: does it make sense to call this tag name? given the resource is apit_tag
```suggestion
* `name` - (Required) The name of the tag. It must be known in the API Management instance. Changing this forces a new API Management API Tag to be created.
```

**Comment by katbyte** [comment]
Looks like we have some test failures:
```
------- Stdout: -------
=== RUN   TestAccApiManagementApiTag_basic
=== PAUSE TestAccApiManagementApiTag_basic
=== CONT  TestAccApiManagementApiTag_basic
    testcase.go:110: Step 1/2 error: Error running apply: exit status 1
        
        Error: assigning to Api "Api Tag: (Tag Name \"acctest-Api-Tag-211228181815017528\" / Api Name \"acctestapi-211228181815017528\" / Service Name \"acctestAM-211228181815017528\" / Resource Group \"acctestRG-211228181815017528\")": apimanagement.TagClient#AssignToAPI: Failure responding to request: StatusCode=404 -- Original Error: autorest/azure: Service returned an error. Status=404 Code="ResourceNotFound" Message="Tag not found."
        
          with azurerm_api_management_api_tag.test,
          on terraform_plugin_test.tf line 34, in resource "azurerm_api_management_api_tag" "test":
          34: resource "azurerm_api_management_api_tag" "test" {
        
--- FAIL: TestAccApiManagementApiTag_basic (316.84s)
FAIL
```

**Comment by meilz381** [comment]
Thank you @katbyte for your hints and reviews. I hopefully fixed the test failures.

**Comment by katbyte** [comment]
no worries! thats why we run them before merge 🙂 doing that again now

**Comment by katbyte** [comment]
Another test failure:
```
------- Stdout: -------
=== RUN   TestAccApiManagementApiTag_requiresImport
=== PAUSE TestAccApiManagementApiTag_requiresImport
=== CONT  TestAccApiManagementApiTag_requiresImport
    testcase.go:110: Step 2/2, expected an error with pattern, no match on: Error running pre-apply refresh: exit status 1
        
        Error: ID contained more segments than required: "/subscriptions/*******/resourceGroups/acctestRG-220105142222439592/providers/Microsoft.ApiManagement/service/acctestAM-220105142222439592/apis/acctestapi-220105142222439592/tags/acctest-Tag-220105142222439592", map[tags:acctest-Tag-220105142222439592]
        
          with azurerm_api_management_api_tag.import,
          on terraform_plugin_test.tf line 46, in resource "azurerm_api_management_api_tag" "import":
          46:   api_id = azurerm_api_management_api_tag.test.id
        
--- FAIL: TestAccApiManagementApiTag_requiresImport (353.74s)
FAIL
```

**Comment by meilz381** [comment]
@katbyte hopefully I managed to run and fix not only the basic test this time. 😅

**Review by katbyte — APPROVED** [review]
Thanks @meilz381 - LGTM 💯

**Pull request merged**